### ISSUE TITLE
[Android] Tooltip are not read by the screen reader

### ISSUE THREAD

**Issue by mattia-venturini** [flutter/flutter#162509]
### Steps to reproduce

Add a tooltip on an IconButton or a FloatingActionButton, or wrap an Icon with a Tooltip widget (check the example below).
Open the app on Android with  the screen reading enabled (I am using TalkBack).

### Expected results

I expect the screen reader to read the tooltip message I provided, when focusing on one of the buttons.
I would also expect the Icon to e focusable, and to be read as its tooltip.

Side note: the tooltip are working as expected when I long-press the elements.

### Actual results

The tooltips are all ignored.
The screen reader simply reads "Button" when focusing on both the IconButton and the FloatingActionButton. The Icon with the tooltip is simply skipped.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(AccessibilityExample());
}

class AccessibilityExample extends StatelessWidget {
  const AccessibilityExample({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('Example'),
          actions: [
            IconButton(
              icon: Icon(Icons.settings),
              tooltip: 'Settings',
              onPressed: () {},
            )
          ],
        ),
        body: Center(
          child: Tooltip(
            message: 'Will this tooltip work?',
            child: Icon(
              Icons.accessibility,
              size: 40,
              color: Colors.blue,
            ),
          ),
        ),
        floatingActionButton: FloatingActionButton(
          tooltip: 'Confirm',
          onPressed: () {},
          child: Icon(Icons.check),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.27.2, on macOS 15.2 24C101 darwin-x64, locale it-IT)
    • Flutter version 3.27.2 on channel stable at /Users/mattiaventurini/fvm/versions/3.27.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 68415ad1d9 (3 weeks ago), 2025-01-13 10:22:03 -0800
    • Engine revision e672b006cb
    • Dart version 3.6.1
    • DevTools version 2.40.2

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/mattiaventurini/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11609105)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11609105)

[✓] VS Code (version 1.96.4)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.102.0

[✓] Connected device (3 available)
    • SM T590 (mobile) • 86b8d4d4 • android-arm    • Android 10 (API 29)
    • macOS (desktop)  • macos    • darwin-x64     • macOS 15.2 24C101 darwin-x64
    • Chrome (web)     • chrome   • web-javascript • Google Chrome 132.0.6834.159

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by maheshj01** [comment]
Hello @mattia-venturini, Thanks for filing the issue. Inorder for the tooltip semantics to work you need to add
`excludeFromSemantics: false,` and for the icons semantic to be read  you need to add `semanticLabel` other wise it will be skipped

```dart
 Center(
    child: Tooltip(
      excludeFromSemantics: false, 
      message: 'Will this tooltip work?',
      child: Icon(
        semanticLabel: 'Accessibility',
        Icons.accessibility,
        size: 40,
        color: Colors.blue,
      ),
    ),
),
```

But At this time the tooltip semantics are being read even when not visisble this is currently being tracked here https://github.com/flutter/flutter/issues/108829.

Closing as this is working as intended. Incase you disagree please feel free to write in the comments and we will reopen it.

**Comment by mattia-venturini** [comment]
Thank you for your reply. I did not know about the `semanticLabel` parameter.

Now the screen reader reads both the `semanticsLabel` and the `message` (or the `tooltip` in case of the IconButton). If they are the same the tooltip is ignored. So the tooltip is ignored if its child doesn't have a semantic itself. If this is the expected behavior, I think it should be better documented in the [Tooltip documentation](https://api.flutter.dev/flutter/material/Tooltip-class.html).
